riscv/csrs.cc

```diff
@@ -1574,3 +1574,66 @@ void jvt_csr_t::verify_permissions(insn_t insn, bool write) const {
     }
   }
 }
+
+virtualized_indirect_csr_t::virtualized_indirect_csr_t(processor_t* const proc, csr_t_p orig, csr_t_p virt):
+  virtualized_csr_t(proc, orig, virt) {
+}
+
+void virtualized_indirect_csr_t::verify_permissions(insn_t insn, bool write) const {
+  csr_t::verify_permissions(insn, write);
+  if (state->v)
+    virt_csr->verify_permissions(insn, write);
+  else
+    orig_csr->verify_permissions(insn, write);
+}
+
+sscsrind_reg_csr_t::sscsrind_reg_csr_t(processor_t* const proc, const reg_t addr, const reg_t init, csr_t_p iselect) :
+  basic_csr_t(proc, addr, init),
+  iselect(iselect) {
+}
+
+void sscsrind_reg_csr_t::verify_permissions(insn_t insn, bool write) const {
+  // Don't call base verify_permission for VS registers because the addresses always fail
+  // the permission check when remapped to S-mode
+  if (!(address >= CSR_VSIREG && address <= CSR_VSIREG6))
+    basic_csr_t::verify_permissions(insn, write);
```

Comment:
Great catch -- fixed now. If VS mode access is redirected, the CSR address won't match the instruction, but if it tries to access it directly, it will match, so we need to prevent that.